worlds/dlcquest/Rules.py

```diff
@@ -215,25 +214,34 @@ def has_coin(state, player: int, coins: int):
                  lambda state: state.has("Cut Content Pack", player))
         set_rule(world.get_entrance("Blizzard", player),
                  lambda state: state.has("Season Pass", player))
-        set_rule(world.get_entrance("Boss Door", player),
-                 lambda state: state.has("Big Sword Pack", player) and state.has("Really Big Sword Pack",
-                                                                                 player) and state.has(
-                     "Unfathomable Sword Pack", player))
         set_rule(world.get_location("I Get That Reference!", player),
                  lambda state: state.has("Death of Comedy Pack", player))
         set_rule(world.get_location("Story is Important", player),
                  lambda state: state.has("DLC NPC Pack", player))
         set_rule(world.get_entrance("Pickaxe Hard Cave", player),
                  lambda state: state.has("Pickaxe", player))
+        big_sword_location = [world.get_location("Big Sword Pack", player)]
+        big_sword_self = item_name_in_locations("Big Sword Pack", player, big_sword_location)
+        really_big_sword_location = [world.get_location("Big Sword Pack", player),
+                                     world.get_location("Really Big Sword Pack", player)]
+        really_big_sword_self = item_name_in_locations("Really Big Sword Pack", player, really_big_sword_location)
+        unfathomable_sword_location = [world.get_location("Big Sword Pack", player),
+                                       world.get_location("Really Big Sword Pack", player),
+                                       world.get_location("Unfathomable Sword Pack", player)]
+        unfathomable_sword_self = item_name_in_locations("Unfathomable Sword Pack", player, unfathomable_sword_location)
```

Comment:
I don't understand why this is required and there is no commend in code nor in PR. I would prefer having a comment there explaining what it is does.

Anyway, I think it would be better to add to the list of locations than to look them up again to make it faster and easier on the eyes.
ie.
```python
sword_locations =  [world.get_location(...)]
big_sword_self = ...(..., sword_locations)
sword_locations.append(world.get_location(...))
really_big_sword_self = ...(..., sword_locations)
...
```

Reply:
I can't speak for the quality of the code (and for the record, I absolutely agree with your suggestion).

But for the "why" this is required, it is because in DLC Quest, the last 3 DLCs (these 3 swords) and bought during the boss fight, and necessary to fight the boss. So in this case, the logic will not allow the player to go battle the boss if they did not receive all 3 swords, as that would lead to a softlock during the boss, due to the way the fight events are done (The boss comes down, offers you a DLC, then just waits here indefinitely until you use the newly acquired sword on him. If buying the DLC sends a check and you can't get the sword, you would need to alt-f4 your game to escape).

But we added one exception to this rule: You can start the boss fight if you do not have one of these mandatory swords, if and only if these swords will be obtained during the fight itself, before or during the specific event that requires it, as then, you would not get softlocked.

So sword1 can be obtained before the fight, or in the sword1 location. Sword2 can be obtained before the fight, in sword1, or in sword2. And so on.

The mod behavior matches this logic, locking the boss door until you have all 3 swords, but it scouts the 3 boss locations and might let you in, in these same conditions.

Reply:
oh, that makes complete sense. i would add a comment like
`# allow required swords to be obtained during boss fight`
or something

Reply:
Fixed, plus a lot of cleanup